### PR TITLE
Admin

### PR DESCRIPTION
**Pull request by MatthewOakley**
I did some basic work of getting some of the stats on the admin page like the number of users and conversations. I was going to add test cases but worked off of previous code that has test cases built into it. Let me know if you have any questions :)

### CODE REVIEW

File: src/main/java/codeu/controller/AdminServlet.java [hunk 1]
```diff
@@ -1,5 +1,13 @@
 package codeu.controller;
 
+/** Getting the user stats for the webpage. */
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+
+/** Getting the conversation stats for the webpage */
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: You don't need comments for imports, you can delete these

File: src/main/java/codeu/controller/AdminServlet.java [hunk 2]
```diff
@@ -22,19 +30,55 @@
    * the database and checking to make sure its valid and not null
    * and sending this information to the webpage
    */
+   
+  /** Store class that gives access to Users. */
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: For single line comments like these you can use `//` style. The only time we require `/** */` style comments is for docstrings which are exclusive to classes and functions.

File: src/main/java/codeu/controller/AdminServlet.java [hunk 3]
```diff
@@ -22,19 +30,55 @@
...
+  private UserStore userStore;
+  
+  /** Store class that gives access to Conversations. */
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: `//`

File: src/main/java/codeu/controller/AdminServlet.java [hunk 4]
```diff
@@ -22,19 +30,55 @@
...
+  private ConversationStore conversationStore;
   
   /** This is for the inital setup of the admin page */
   @Override
   public void init() throws ServletException {
     super.init();
+    
+    /** This is for getting the current instance of users */
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: `//`

File: src/main/java/codeu/controller/AdminServlet.java [hunk 5]
```diff
@@ -22,19 +30,55 @@
...
+    setUserStore(UserStore.getInstance());
+    
+    /** This is for getting the current instance of conversations */
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: `//`

File: src/main/java/codeu/controller/AdminServlet.java [hunk 6]
```diff
@@ -22,19 +30,55 @@
...
+    setConversationStore(ConversationStore.getInstance());
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: A note on comments: I only recommend them when it's not immediately clear to someone reading your code what the purpose of your code is / what is going on. Here, it's pretty obvious. Therefore, I'd say it's actually better to let your code speak for itself here and delete them.

I'd go through some of your other comments as well and evaluate whether you need them. In college they'll make you comment `*everything*`, but, in reality, it's not always the case in industry.

File: src/main/java/codeu/controller/AdminServlet.java [hunk 7]
```diff
@@ -22,19 +30,55 @@
...
   }
   
   /** when user visits this sends them to admin page */
   @Override
   public void doGet(HttpServletRequest request, 
       HttpServletResponse response)
       throws IOException, ServletException {
-    String requestUrl = request.getRequestURI();
+    
+    /** This gets the current number of users in the database*/
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: Space after database

File: src/main/java/codeu/model/store/basic/ConversationStore.java [hunk 8]
```diff
@@ -100,4 +100,13 @@ public Conversation getConversationWithTitle(String title) {
   public void setConversations(List<Conversation> conversations) {
     this.conversations = conversations;
   }
+  
+  /** Returns the amount of conversations currently in the database. */
+  public int getConversationAmount(){
+    if(this == null){
+      return 0;
+    }
```
Comment (erschwartz) [inline src/main/java/codeu/model/store/basic/ConversationStore.java]: Fix the spacing in this method please. Space always after () and in if statements

```java
public int getConversationAmount() {
  if (this == null) {
    return 0;
  }
}

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 9]
```diff
@@ -126,5 +126,14 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+  
+  /** Returns the amount of users currently registrated. */
+  public int getUserAmount(){
```
Comment (erschwartz) [inline src/main/java/codeu/model/store/basic/UserStore.java]: Fix spacing here please. See comment above.

File: src/main/java/codeu/model/store/basic/ConversationStore.java [hunk 10]
```diff
@@ -100,4 +100,13 @@ public Conversation getConversationWithTitle(String title) {
   public void setConversations(List<Conversation> conversations) {
     this.conversations = conversations;
   }
+  
+  /** Returns the amount of conversations currently in the database. */
+  public int getConversationAmount(){
+    if(this == null){
```
Comment (erschwartz) [inline src/main/java/codeu/model/store/basic/ConversationStore.java]: I think this check is unnecessary as you can assume that they're calling this method on an instance that is obtained/constructed via getInstance

File: src/main/java/codeu/model/store/basic/UserStore.java [hunk 11]
```diff
@@ -126,5 +126,14 @@ public boolean isUserRegistered(String username) {
...
+    if(this == null){
```
Comment (erschwartz) [inline src/main/java/codeu/model/store/basic/UserStore.java]: See comment above

**Review by erschwartz — CHANGES_REQUESTED** [review]
Nice progress, Matt!

**Review by StevenAb — APPROVED** [review]
Apart from what Erika mentioned, this LGTM!

**Comment by erschwartz** [comment]
Hi Matt, if you finished the changes for this, make sure to push them up so we can see them.

File: src/main/java/codeu/controller/AdminServlet.java [hunk 12]
```diff
@@ -22,19 +29,70 @@
...
+  /**
+   * Sets the MessageStore used by this servlet. 
+   * This function provides a common setup method
+   * for use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
   }
   
   /** when user visits this sends them to admin page */
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: when -> When

File: src/main/java/codeu/controller/AdminServlet.java [hunk 13]
```diff
@@ -22,19 +29,70 @@
...
   @Override
   public void doGet(HttpServletRequest request, 
       HttpServletResponse response)
       throws IOException, ServletException {
-    String requestUrl = request.getRequestURI();
+    
+    /** This gets the current number of users in the database */
```
Comment (erschwartz) [inline src/main/java/codeu/controller/AdminServlet.java]: Please see the comment I made before about comments. Single line comments that are not docstrings can use // and you can delete these as they are unnecessary

**Review by erschwartz — CHANGES_REQUESTED** [review]

**Review by erschwartz — APPROVED** [review]
LGTM

**Pull request merged**